fedimint-server/src/config/distributedgen.rs

```diff
@@ -0,0 +1,480 @@
+use std::collections::{BTreeMap, HashMap};
+use std::fmt::Debug;
+use std::hash::Hash;
+use std::io::Write;
+
+use anyhow::{ensure, format_err};
+use bitcoin_hashes::sha256::Hash as Sha256;
+use bitcoin_hashes::sha256::HashEngine;
+use fedimint_api::config::{DkgGroup, DkgMessage, DkgPeerMsg, ISupportedDkgMessage};
+use fedimint_api::core::ModuleInstanceId;
+use fedimint_api::net::peers::MuxPeerConnections;
+use fedimint_api::BitcoinHash;
+use fedimint_api::PeerId;
+use hbbft::crypto::poly::Commitment;
+use hbbft::crypto::{G1Projective, G2Projective, PublicKeySet, SecretKeyShare};
+use rand::{CryptoRng, RngCore};
+use serde::de::DeserializeOwned;
+use serde::Serialize;
+use tbs::hash::hash_bytes_to_curve;
+use tbs::poly::Poly;
+use tbs::Scalar;
+use threshold_crypto::serde_impl::SerdeSecret;
+
+use crate::*;
+
+struct Dkg<G> {
+    gen_g: G,
+    peers: Vec<PeerId>,
+    our_id: PeerId,
+    threshold: usize,
+    f1_poly: Poly<Scalar, Scalar>,
+    f2_poly: Poly<Scalar, Scalar>,
+    hashed_commits: BTreeMap<PeerId, Sha256>,
+    commitments: BTreeMap<PeerId, Vec<G>>,
+    sk_shares: BTreeMap<PeerId, Scalar>,
+    pk_shares: BTreeMap<PeerId, Vec<G>>,
+}
+
+/// Implementation of "Secure Distributed Key Generation for Discrete-Log Based Cryptosystems"
```

Comment:
I don' `impl` blocks can have docstrings. This should be on `struct`? Does this work? How is this not triggering the compiler. What is happening? I need a coffee.